platforms/BUILD

```diff
@@ -1,25 +1,38 @@
 package(default_visibility = ["//visibility:public"])
 
-config_setting(
-    name = "linux_x86_64",
+platform(
+    name = "linux_amd64",
```

Comment:
`s/amd64/x86_64/g` to be consistent with bazel's `cpu` constraint naming? and also so that the config_setting names match the platform names

I get that the Arch exec property doesn't match, but IMO that's more of a quirk in our RE API and we should stick with bazel's conventions

Reply:
I wasn't sure on what to use so I was fiddling around with both options and `amd64` was slightly shorter to write.

Will change to `x86_64`.